src/display/editor/tools.js

```diff
@@ -826,51 +889,25 @@ class AnnotationEditorUIManager {
     return false;
   }
 
-  /**
-   * Unselect the current editor.
-   */
-  unselect() {
-    if (this.#activeEditor) {
-      this.#activeEditor.parent.setActiveEditor(null);
-    }
-  }
-
   /**
    * Delete the current editor or all.
    */
   delete() {
-    let cmd, undo;
-    if (this.#isAllSelected) {
-      this.#previousActiveEditor = this.#activeEditor = null;
-      const editors = Array.from(this.#allEditors.values());
-      cmd = () => {
-        for (const editor of editors) {
-          if (!editor.isEmpty()) {
-            editor.remove();
-          }
-        }
-      };
-
-      undo = () => {
-        for (const editor of editors) {
-          this.#addEditorToLayer(editor);
-        }
-      };
+    if (!this.hasSelection) {
+      return;
+    }
 
-      this.addCommands({ cmd, undo, mustExec: true });
-    } else {
-      if (!this.#activeEditor) {
-        return;
-      }
-      const editor = this.#activeEditor;
-      this.#previousActiveEditor = this.#activeEditor = null;
-      cmd = () => {
+    const editors = [...this.#selectedEditors];
```

Comment:
*This __potential__ issue exists in the old code as well, but it's something that I just noticed:*

By the time that the `cmd` or `undo` functions run, is it possible the `#selectedEditors` has changed (e.g. it could have become empty) such that the constant `editors`-Array created here is no longer accurate?
So, should we perhaps instead be iterating through `this.#selectedEditors` in the loops below?

Reply:
Let's do the following:
 - create 2 editors A & B;
 - select all;
 - delete
 - undo

After deleting, `#selectedEditors` is empty (nothing is selected!) then if you undo in iterating on `#selectedEditors` then you'll get nothing.